File: step_definitions/Integration_steps.js
const { Given, When, Then } = require('@cucumber/cucumber');
const LoginPage = require('../pages/LoginPage');
const DashboardPage = require('../pages/DashboardPage');

Given('the user is on the login page', async function () {
  await LoginPage.open();
});

When('the user enters valid credentials', async function () {
  await LoginPage.enterCredentials('username', 'password');
});

When('the user clicks the login button', async function () {
  await LoginPage.clickLogin();
});

Then('the user should be redirected to the dashboard', async function () {
  await DashboardPage.waitForDashboard();
});

Then('the dashboard should display the user\'s profile information', async function () {
  await DashboardPage.verifyUserProfile();
});

File: features/step_definitions/Integration_steps.rb
# # features/step_definitions/login_steps.rb

# # require_relative '../pages/login_page'
# # require_relative '../pages/dashboard_page'

# Given('the user is on the login page') do
#   LoginPage.open
# end

# When('the user enters valid credentials') do
#   LoginPage.enter_credentials('username', 'password')
# end

# When('the user clicks the login button') do
#   LoginPage.click_login
# end

# Then('the user should be redirected to the dashboard') do
#   DashboardPage.wait_for_dashboard
# end

# Then('the dashboard should display the user\'s profile information') do
#   DashboardPage.verify_user_profile
# end

# Then('this step always passes') do
#   pending 'This step always passes'
# end